[FEAT]: Firebase 연동을 위한 설정 파일 추가

joonamin opened the pull request:
### 개요
추후에 OAuth2 + 회원가입 인증 로직을 위한 Configuration 파일들입니다.
### 변경사항
--
### 관련 지라 및 위키 링크
BDBD-354 
### 리뷰어에게 하고 싶은 말
Firebase 인증과 관련된 커스텀 필터는 설정 파일 PR이 Accept된다면 추가하여 올리겠습니다. 부족한 점 지적은 항상 환영입니다 :)

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

minseonglove reviewed (approved):
주석을 이쁘게 잘 해놨군요

onikatashi commented:
멋있습니다. 다만 yaml 파일에 빈공간이 생긴거는 초큼 불편하네요 ㅎㅎ

konempty commented on src/main/resources/application.yaml:
```diff
@@ -19,3 +19,6 @@ spring:
       matching-strategy: ant_path_matcher
+
+
+
```
불필요한 개행이 조금 많아보이네요 ㅎㅎ..

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/config/SecurityConfig.java:
```diff
@@ -0,0 +1,58 @@
+        // 1. static resource에 대한 Filter 적용 X
+        // 2. 로그인, 회원가입에 대한 Filter 적용 X
+        web.ignoring()
+                .antMatchers("/resources/static/**")
```
저희 리소스중에 이런 url을 사용하는 리소스가 있었나요??

joonamin replied:
추후에 프론트에서 페이지를 요청할 때, 정적 리소스들에 대해서는 별다른 인증을 필요로 하지 않는다는게 보장되어서 미리 작성하였습니다 !

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/config/SecurityConfig.java:
```diff
@@ -0,0 +1,58 @@
+    /*
+        아래는 Custom Filter의 매개변수로 넘길 예정
+     */
+    @Autowired
```
오... Autowired는 지양해야해요 잘못하면 꼬이기 쉽거든요 ㅎㅎ... 
대신 생성자를 통한 주입을 저는 주로 쓰는것 같아요!

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/config/FirebaseConfig.java:
```diff
@@ -0,0 +1,30 @@
+    private final String firebaseSdkPath = "./firebase.json";
+
+    @Bean
+    public FirebaseAuth firebaseAuth() throws IOException {
```
👍 👍 👍

konempty commented on .gitignore:
```diff
@@ -4,6 +4,7 @@ build/
 !gradle/wrapper/gradle-wrapper.jar
 !**/src/main/**/build/
 !**/src/test/**/build/
+firebase.json
```
👍

konempty commented on build.gradle:
```diff
@@ -40,6 +40,8 @@ dependencies {
     implementation group: 'com.mortennobel', name: 'java-image-scaling', version: '0.8.6'
     implementation fileTree(dir: 'libs', include: ['*.jar'])
 
+    implementation group: 'com.google.firebase', name: 'firebase-admin', version: '8.0.1'
```
흐음 아무래도 short방식으로 통일하는게 깔끔할것 같네요! 요 라인을 short형식으로 바꾸면서 java-image-scaling 이것도 short 형식으로 통일시켜 주세요 :)

konempty commented:
> Kudos, SonarCloud Quality Gate passed!    [![Quality Gate passed](https://camo.githubusercontent.com/e4a90cf346b46d44c5652111d68603714bfb7693043d07a95992de3a30bfce04/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f5175616c6974794761746542616467652f7061737365642d313670782e706e67)](https://sonarcloud.io/dashboard?id=FakeDevelopers_BidderBidder_BE&pullRequest=22)
> 
> [![Bug](https://camo.githubusercontent.com/4c6102327f5a954f9c8acaf2e2714183157a9e41717b371b2cd585cf25057310/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f6275672d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=BUG) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=BUG) [![Vulnerability](https://camo.githubusercontent.com/3ba1ee49636ffc3427e38649a9f8a65ee392f28e8a662fcf96ce24cefbb520e9/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f76756c6e65726162696c6974792d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_Bid



> Kudos, SonarCloud Quality Gate passed!    [![Quality Gate passed](https://camo.githubusercontent.com/e4a90cf346b46d44c5652111d68603714bfb7693043d07a95992de3a30bfce04/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f5175616c6974794761746542616467652f7061737365642d313670782e706e67)](https://sonarcloud.io/dashboard?id=FakeDevelopers_BidderBidder_BE&pullRequest=22)
> 
> [![Bug](https://camo.githubusercontent.com/4c6102327f5a954f9c8acaf2e2714183157a9e41717b371b2cd585cf25057310/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f6275672d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=BUG) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=BUG) [![Vulnerability](https://camo.githubusercontent.com/3ba1ee49636ffc3427e38649a9f8a65ee392f28e8a662fcf96ce24cefbb520e9/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f76756c6e65726162696c6974792d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=VULNERABILITY) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=VULNERABILITY) [![Security Hotspot](https://camo.githubusercontent.com/fb735cbe76f8d5e1679c76ce83b740ceb1eaf62de4f7bf88623dc9953261aff7/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f73656375726974795f686f7473706f742d313670782e706e67)](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=SECURITY_HOTSPOT) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=SECURITY_HOTSPOT) [![Code Smell](https://camo.githubusercontent.com/8fe18b2dfb6f7d4e44582f281b29f617eb5ae07c248d2002ca586e91da219212/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f636f64655f736d656c6c2d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=CODE_SMELL) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=CODE_SMELL) [3 Code Smells](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&resolved=false&types=CODE_SMELL)
> 
> [![No Coverage information](https://camo.githubusercontent.com/86785eaa572e061f5ecb3f8bf1cbb7b8eca44925083f92d067805b4362cead0c/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f436f76657261676543686172742f4e6f436f766572616765496e666f2d313670782e706e67)](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_BE&pullRequest=22) No Coverage information [![0.0%](https://camo.githubusercontent.com/8047c63e1f9ed03f63001e1eadce4676bade3e0f83ec690a9c625287796248a6/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f4475706c69636174696f6e732f332d313670782e706e67)](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_BE&pullRequest=22&metric=new_duplicated_lines_density&view=list)

코드스멜 수정 부탁드려요!

konempty reviewed (changes requested):
크흐 고생하셨습니다 👍 
조금만더 고생부탁드릴게요! 수정하고 알려주세요 :)

joonamin committed:
CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

joonamin committed:
CHORE : `BDBD-354` 코드 스멜 제거

joonamin committed:
CHORE : `BDBD-354` 코드 스멜 제거 -2

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/config/SecurityConfig.java:
```diff
@@ -0,0 +1,56 @@
+    @Autowired
+    private FirebaseAuth firebaseAuth; // Firebase 토큰 정보
+    @Autowired
+    private UserDetailsService userDetailsService;
```
Autowired 는 취향이니까 넘어갈게요! 하지만 부작용만 확실히 알고 쓰시면 좋을것 같아요 :)

joonamin replied:
실제 로직을 작성할 때는 생성자를 통해서 가져올 예정입니다 :)
일단 설정 파일 작성 때는 @Autowired가 좀 더 직관적이여서 목업 느낌으로 작성했습니다!

konempty reviewed (approved):
고생하셨습니다 👍
